### DISCUSSION TITLE
How do you listen to right click events on an element?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [zauberzeug/nicegui#800] [Q&A] [answered]
### Question

The vue way of reading rmb events seems to be using `@click.right` https://vuejs.org/guide/essentials/event-handling.html#event-modifiers, cant quite seem to get that to work with `Element.on(''click.right", lambda: print("cb"))`. Am I doing something wrong?

Would also like to be able to cancel the browser default RMB menu.

While Im well here I might as well ask for double click & if its possible to listen to keyboard presses on a per element basis (so, I only want to receive events if the element the event belongs to is hovered while a key is pressed).

**Comment by falkoschindler** [comment, marked as answer]
I'm not sure if "right" is a valid event modifier. But apart from that it looks like the `contextmenu` event gets in the way, such that even a normal `click` event is not emitted when using the right mouse button.

But you can subscribe to a `contextmenu` event and prevent the default behavior like this:
```py
ui.label('Right click me').on('contextmenu.prevent', lambda: ui.notify('Right!'))
```

And handling double clicks can be done using the `dblclick` event:
```py
ui.label('Double click me').on('dblclick', lambda: ui.notify('Double!'))
```

**Reply by ItsCubeTime** [reply]
Thanks! What are the docs I should be looking at to see all the possible events?

**Reply by falkoschindler** [reply]
I tend to look into the [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event). But there are probably other good references about JavaScript events.
And [here](https://nicegui.io/documentation#generic_events) is our documentation of possible event modifiers.

**Reply by kfalcami** [reply]
FYI nicegui generic events doc page has moved to https://nicegui.io/documentation/generic_events
